localstack/aws/handlers/internal_requests.py

```diff
@@ -0,0 +1,29 @@
+import logging
+from types import MappingProxyType
+
+from localstack.http import Response
+
+from ..api import RequestContext
+from ..chain import Handler, HandlerChain
+from ..connect import INTERNAL_REQUEST_PARAMS_HEADER, load_dto
+
+LOG = logging.getLogger(__name__)
+
+
+class InternalRequestParamsEnricher(Handler):
+    """
+    This handler sets the internal call DTO in the request context.
+
+    Important: This must be invoked after account and region enrichers because
+    it may override them.
+    """
+
+    def __call__(self, chain: HandlerChain, context: RequestContext, response: Response):
+        if header := context.request.headers.get(INTERNAL_REQUEST_PARAMS_HEADER):
+            try:
+                dto = MappingProxyType(load_dto(header))
+            except Exception as e:
+                LOG.error("Error loading request parameters '%s', Error: %s", header, e)
```

Comment:
Maybe we should use `LOG.exception` here? Not being able to parse the internal request params seems to be a weird internal error which should get some visibility.

Reply:
Good point, will do!